src/test/java/org/broadinstitute/hellbender/utils/UtilsUnitTest.java

```diff
@@ -978,4 +981,78 @@ public void testTestFilterCollectionByExpressions(Set<String> values, Collection
         Set<String> actual = Utils.filterCollectionByExpressions(values, filters, exactMatch);
         Assert.assertEquals(actual, expected);
     }
+
+    @DataProvider(name = "concatMultipleByteArraysData")
+    public Object[][] concatMultipleByteArraysData() {
+        return new Object[][]{
+                // expected, arrays
+                new Object[]{ ArrayUtils.EMPTY_BYTE_ARRAY, new byte[] { } },
+                new Object[]{ new byte[] {100, 100}, new byte[] {100, 100} },
+                new Object[]{ new byte[] {100, 100}, new byte[] {100, 100}, new byte[] { } },
+                new Object[]{ ArrayUtils.EMPTY_BYTE_ARRAY, new byte[] { }, new byte[] { } },
+                new Object[]{ new byte[] {100, 100, 100, 100}, new byte[] {100, 100}, new byte[] {100, 100} },
+        };
+    }
+
+    // expected listed first to accomodate varargs arrays having to be last parameter
+    @Test(dataProvider = "concatMultipleByteArraysData")
+    public void testConcatMultipleByteArrays(byte[] expected, byte[] ... arrays) {
+        Assert.assertEquals(Utils.concat(arrays), expected);
+    }
+
+    @Test
+    public void testConcatMultipleWithNoArrays(){
+        Assert.assertEquals(Utils.concat(), ArrayUtils.EMPTY_BYTE_ARRAY);
+    }
+
+    @DataProvider(name = "concatTwoByteArraysData")
+    public Object[][] concatTwoByteArraysData() {
+        return new Object[][]{
+                // arr1, arr2, expected
+                new Object[]{ new byte[] { }, new byte[] { }, ArrayUtils.EMPTY_BYTE_ARRAY },
+                new Object[]{ new byte[] {100, 100}, new byte[] { }, new byte[] {100, 100} },
+                new Object[]{ new byte[] { }, new byte[] {100, 100}, new byte[] {100, 100} },
+                new Object[]{ new byte[] {100, 100}, new byte[] {100, 100}, new byte[] {100, 100, 100, 100} },
+        };
+    }
+
+    @Test(dataProvider = "concatTwoByteArraysData")
+    public void testConcatTwoByteArrays(byte[] arr1, byte[] arr2, byte[] expected) {
+        Assert.assertEquals(Utils.concat(arr1, arr2), expected);
+    }
+
+    @DataProvider(name = "concatAnyTypeArraysData")
+    public Object[][] concatAnyTypeArraysData() {
+        return new Object[][]{
+                // arr1, arr2, constructor, expected
+                new Object[]{ new Integer[] { }, new Integer[] { }, (IntFunction<Integer[]>)Integer[]::new,
+                        new Integer[] { } },
+                new Object[]{ new Integer[] {1, 2, 3}, new Integer[] { }, (IntFunction<Integer[]>)Integer[]::new,
+                        new Integer[] {1, 2, 3} },
+                new Object[]{ new Integer[] { }, new Integer[] {4, 5, 6}, (IntFunction<Integer[]>)Integer[]::new,
+                        new Integer[] {4, 5, 6} },
+                new Object[]{ new Integer[] {1, 2, 3}, new Integer[] {4, 5, 6}, (IntFunction<Integer[]>)Integer[]::new,
+                        new Integer[] {1, 2, 3, 4, 5, 6} },
+        };
```

Comment:
Comments on the test cases for the generic version of concat():

- Include an additional test case with { 4, 5, 6 } and { 1, 2, 3 } passed in the reverse order, giving { 4, 5, 6, 1, 2, 3 }. This will demonstrate that the method is preserving input order correctly and not doing something silly like sorting the values during concatenation.